pkg/clusterconditions/promql/promql.go

```diff
@@ -69,7 +84,11 @@ func (p *PromQL) Valid(ctx context.Context, condition *configv1.ClusterCondition
 // false when the PromQL evaluates to 0, and an error if the PromQL
 // returns no time series or returns a value besides 0 or 1.
 func (p *PromQL) Match(ctx context.Context, condition *configv1.ClusterCondition) (bool, error) {
-	clientConfig := api.Config{Address: p.Address}
+	address, err := p.Address(ctx)
```

Comment:
lookup is done every match attempt in case of delete and recreate to get different IP.